### DISCUSSION TITLE
Should build fail if `publishRelease` fail because Sentry instance is down ?

### DISCUSSION THREAD

**Discussion by mrleblanc101** [nuxt-community/sentry-module#585] [Q&A] [answered]
Hi,
Should build fail if `publishRelease` fail because Sentry instance is down ?
I'm trying to deploy my app, and our self-hosted Sentry is currently down.
I can't deploy because I have `publishRelease: true` which fails.
Should we have an option to ignore this is complete the build ?

**Comment by rchl** [comment]
I think you might be able to do that already using the `publishRelease.errorHandler` option. It's part of Sentry own types and is documented as:

```ts
    /**
     * When a CLI error occurs, the plugin will call this function.
     *
     * By default, it will call `invokeErr()`, thereby stopping Webpack
     * compilation. To allow compilation to continue and log a warning instead,
     * set this to
     *   (err, invokeErr, compilation) => {
     *     compilation.warnings.push('Sentry CLI Plugin: ' + err.message)
     *   }
     *
     * Note: `compilation` is typed as `unknown` in order to preserve
     * compatibility with both Webpack 4 and Webpack 5 types, If you need the
     * correct type, in Webpack 4 use `compilation.Compilation` and in Webpack 5
     * use `Compilation`.
     */
    errorHandler?: (
      err: Error,
      invokeErr: () => void,
      compilation: unknown
    ) => void;
```

BTW. There is also this Nuxt issue that causes Sentry errors during publishing to be hidden: https://github.com/nuxt/nuxt/issues/21284
Not sure if you were also hit by this and had to guess what the issue is when it happened?

**Comment by mrleblanc101** [comment]
@rchl Thanks for the tip, will try !
The error seems to work for me:
![Capture d’écran, le 2023-06-12 à 10 17 58](https://github.com/nuxt-community/sentry-module/assets/17392251/a77949cf-0b42-4d70-9481-1df6e3bf1e7c)
Which is indeed caused by the fact that our self-hosted Sentry is down:
![Capture d’écran, le 2023-06-12 à 10 22 40](https://github.com/nuxt-community/sentry-module/assets/17392251/f10f962a-399c-4e85-abc3-edd732c3ba2b)

**Comment by mrleblanc101** [comment]
I was not able to make it work.
I tried both in my `nuxt.config.js`:
```
sentry: {
    config: {
        errorHandler: (err, invokeErr, compilation) => {
            compilation.warnings.push('Sentry CLI Plugin: ' + err.message);
        },
    },
},
```
And 
```
sentry: {
      errorHandler: (err, invokeErr, compilation) => {
          compilation.warnings.push('Sentry CLI Plugin: ' + err.message);
      },
},
```

**Comment by rchl** [comment, marked as answer]
```
sentry: {
  publishRelease: {
    errorHandler: ...
  }
}
```

**Comment by mrleblanc101** [comment]
@rchl The only possibility I didn't try I guess 😅
Thanks, I searched in the Sentry documentation instead of the `sentry-webpack-plugin` documentation.

**Comment by mrleblanc101** [comment]
Maybe this could be in nuxt/sentry documentation.
Or maybe the default should be to warn instead of stopping the build ?
```
publishRelease: {
    errorHandler: (err, invokeErr, compilation) => {
        compilation.warnings.push('Sentry CLI Plugin: ' + err.message);
    },
},
```

**Reply by rchl** [reply]
I think that hard error is a better default. Otherwise the issue can be missed easily during deploy and it will be hard to figure out why things are not working as expected.

Documenting that is a good idea though. We could add another section for publishing releases in the `Guide` section in https://sentry.nuxtjs.org/ with such practical information.

**Reply by rchl** [reply]
added new documentation page with a blurb that explains that: https://sentry.nuxtjs.org/guide/releases
